deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/iterator/MultipleEpochsIterator.java

```diff
@@ -211,7 +211,7 @@ public void setPreProcessor(org.nd4j.linalg.dataset.api.DataSetPreProcessor preP
      */
     @Override
     public boolean hasNext() {
-        if (newEpoch || (newEpoch && epochs == numEpochs)) {
+        if (newEpoch || epochs == numEpochs) {
             log.info("Epoch " + epochs + ", number of batches completed " + lastBatch);
```

Comment:
Hm, old condition is definitely wrong. if(newEpoch) as the only condition makes more sense here.